src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,559 @@
+ * This array has information for each data point w.r.t shape, colors and on click callback along with
+ * x and y co-ordinates.
+ *
+ * @private
+ * @param {object} target - Object containing the subsets
+ * @returns {Array} List of data point subsets
+ */
+const getDataPointValues = (target) => target.internalValuesSubset;
+/**
+ * Checks the data-set is currently shown in the graph and if the y data-point value is null
+ * If they are then true, false otherwise
+ *
+ * @private
+ * @param {object} shownTargets - graph targets config object
+ * @param {object} value - data point value object
+ * @returns {boolean} true if data point needs to be hidden, false otherwise
+ */
+const shouldHideDataPoints = (shownTargets, value) =>
+    shownTargets.indexOf(value.key) < 0 || value.y === null;
+
+/**
+ * Gets all the circles given in the bubble graph.
+ *
+ * @private
+ * @returns {Array} List of circle nodes in bubble chart
+ */
+const getAllCircleNodes = () => {
+    const circleNodes = [];
+    const bubbleGroup = d3.selectAll(`.${styles.bubbleGraphContent}`);
+    bubbleGroup.each(function() {
+        const node = d3.select(this);
+        const bubbleNode = node.selectAll(`.${styles.point} circle`);
+        bubbleNode.each(function() {
+            const circle = d3.select(this);
+            circleNodes.push(...circle[0]);
+        });
+    });
+    return circleNodes;
+};
+
+/**
+ * Enforces blur state for all the bubbles that is not the one hovered on.
+ * This is provided regardless of whether onClick is present or not.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles form bubble graph
+ * @returns {undefined} - returns nothing
+ */
+const enforceBubbleBlur = (nodes) =>
+    nodes.forEach((ele) => {
+        const node = d3.select(ele);
+        if (node.attr("aria-selected") === "false") {
+            node.classed(styles.bubbleBlur, true);
+        }
+    });
+
+/**
+ * Removes the carbon-bubbleBlur style from all the bubbles to unblur all the bubbles in the bubble graph.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles in the bubbles graph.
+ * @returns {undefined} - returns nothing
+ */
+const removeBubbleBlur = (nodes) =>
+    nodes.forEach((ele) => {
+        const node = d3.select(ele);
+        if (node.attr("aria-selected") === "false") {
+            node.classed(styles.bubbleBlur, false);
+        } else {
+            node.attr("aria-selected", false);
+        }
+    });
+
+/**
+ * Handler for the bubble that is hovered on. It blurs all other bubble in the bubble graph except one which is selected.
+ *
+ * @private
+ * @param {HTMLElement} target - Target element bubble hovered on
+ * @param {string} hoverState - Mouse over or Mouse out
+ * @returns {undefined} - returns nothing
+ */
+const blurActionHandler = (target, hoverState) => {
+    d3.select(target.firstChild).attr("aria-selected", true);
```

Comment:
I think we should have this for the element with the class "carbon-point" instead of the circle itself (i.e. firstChild).

Reply:
No, we can't do that way because we are style the circle which is inside the carbon-point. Initially, I also thought like this but this is not possible.

Reply:
since the properties like "fill-opacity and stroke-opacity" are inside the circle we did it that way selecting the firstChild(circle).